CMSgov/design-system
# [WNMDGS-2972] Swap bold headers with semibold headers on CMSgov theme

## Summary
Follow up PR related to https://github.com/CMSgov/design-system/pull/3227.

Isolated only the semibold header change for CMSgov.

## How to test
Confirm that headers 5XL, 2XL, XL, LG, and MD use the semibold (600) font weight.

Example of 5XL: (using "WhatFont" extension)
<img width="670" alt="image" src="https://github.com/user-attachments/assets/fb9173f4-b702-4d13-9e86-a8defaeebe78">


## Checklist
